packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -28,163 +28,176 @@ import { Context, SetBlockNodes } from './root-container';
+	 */
+	const focusTabbable = () => {
+		// Focus is captured by the wrapper node, so while focus transition
+		// should only consider tabbables within editable display, since it
+		// may be the wrapper itself or a side control which triggered the
+		// focus event, don't unnecessary transition to an inner tabbable.
+		if (
+			document.activeElement &&
+			isInsideRootBlock( ref.current, document.activeElement )
+		) {
+			return;
+		}
 
-		// Block Reordering animation
-		useMovingAnimation(
-			wrapper,
-			isSelected || isPartOfMultiSelection,
-			isSelected || isFirstMultiSelected,
-			enableAnimation,
-			index
-		);
+		// Find all tabbables within node.
+		const textInputs = focus.tabbable
+			.find( ref.current )
+			.filter( isTextField )
+			// Exclude inner blocks and block appenders
+			.filter(
+				( node ) =>
+					isInsideRootBlock( ref.current, node ) &&
+					! node.closest( '.block-list-appender' )
+			);
+
+		// If reversed (e.g. merge via backspace), use the last in the set of
+		// tabbables.
+		const isReverse = -1 === initialPosition;
+		const target =
+			( isReverse ? last : first )( textInputs ) || ref.current;
+
+		placeCaretAtHorizontalEdge( target, isReverse );
+	};
+
+	useEffect( () => {
+		if ( ! isMultiSelecting && ! isNavigationMode && isSelected ) {
+			focusTabbable();
+		}
+	}, [ isSelected, isMultiSelecting, isNavigationMode ] );
+
+	// Block Reordering animation
+	useMovingAnimation(
+		ref,
+		isSelected || isPartOfMultiSelection,
+		isSelected || isFirstMultiSelected,
+		enableAnimation,
+		index
+	);
+
+	useEffect( () => {
+		if ( ! isSelected || isLocked ) {
+			return;
+		}
 
 		/**
-		 * Interprets keydown event intent to remove or insert after block if key
-		 * event occurs on wrapper node. This can occur when the block has no text
-		 * fields of its own, particularly after initial insertion, to allow for
-		 * easy deletion and continuous writing flow to add additional content.
+		 * Interprets keydown event intent to remove or insert after block if
+		 * key event occurs on wrapper node. This can occur when the block has
+		 * no text fields of its own, particularly after initial insertion, to
+		 * allow for easy deletion and continuous writing flow to add additional
+		 * content.
 		 *
 		 * @param {KeyboardEvent} event Keydown event.
 		 */
-		const onKeyDown = ( event ) => {
+		function onKeyDown( event ) {
 			const { keyCode, target } = event;
 
-			if ( props.onKeyDown ) {
-				props.onKeyDown( event );
-			}
-
-			if (
-				keyCode !== ENTER &&
-				keyCode !== BACKSPACE &&
-				keyCode !== DELETE
-			) {
+			if ( keyCode !== ENTER ) {
 				return;
 			}
 
-			if ( target !== wrapper.current || isTextField( target ) ) {
+			if ( target !== ref.current || isTextField( target ) ) {
 				return;
 			}
 
 			event.preventDefault();
 
-			if ( keyCode === ENTER ) {
-				insertDefaultBlock( {}, rootClientId, index + 1 );
-			} else {
-				removeBlock( clientId );
```

Comment:
Why did you remove the "removal" handling? is this handled elsewhere?

Reply:
There's a shortcut for it defined as well.

Reply:
I guess we should adapt the comment and should the "Enter" be handled with a shortcut for consistency?